### PR TITLE
chore: Upgrade Argo to v3.4.17 (fixes #10977)

### LINKED ISSUE #10977: Upgrade to latest Argo 3.4.17
### Feature Area

/area backend

### What feature would you like to see?

Upgrade Argo Workflows dependency to v3.4.17

### What is the use case or pain point?

Includes several security/CVE fixes, and current version is 6 months old.  Just a single minor version bump, so should be low-risk/low-impact otherwise.

### Is there a workaround currently?

N/A

---

Love this idea? Give it a 👍.

### PR DESCRIPTION
**Pull request by gmfrasca**
**Description of your changes:**
Fixes #10977 

(Will need a GCR admin to build and push the argoexec and workflow-controller images for CI to pass)

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: manifests/kustomize/third-party/argo/upstream/manifests/namespace-install/kustomization.yaml [hunk 1]
```diff
@@ -4,16 +4,16 @@ resources:
   - ../base
   - ./argo-server-rbac
   - ./workflow-controller-rbac
-patches:
```
Comment (gmfrasca) [inline manifests/kustomize/third-party/argo/upstream/manifests/namespace-install/kustomization.yaml:7]: this was just updated by https://github.com/kubeflow/pipelines/commit/af085769fbb29a72d1b18c6a47d474e58dbd5577, but these manifests are pulled directly from upstream.  It appears upstream itself updated its manifests to using `patches` in the very next release (v3.5.0), so I can revert these specific changes back to use `patches` (and therefore we will maintain kustomize5 compatibility), or just allow the reversion to happen again at next argo upgrade

**Comment by hbelmiro** [comment]
@rimolive /rerun <workflow-name> isn't working.
Can you please take a look to see if I send the right command?

**Comment by rimolive** [comment]
@hbelmiro `/rerun-all` worked, maybe `/rerun-worklow` only works passing the workflow ID? Or maybe you need to send workflow name with double quotes?

Anyways, this is the GHA I'm using to rerun failed workflows. Maybe you can find the answer to that question. https://github.com/estroz/rerun-actions

**Comment by hbelmiro** [comment]
@rimolive yeah, I tried `/rerun-all` earlier and it worked. But when using the workflow ID it didn't.
Let's see with double-quotes.

/rerun-workflow "KFP Backend Tests"

**Comment by zijianjoy** [comment]
I have uploaded new argo images: gcr.io/ml-pipeline/workflow-controller:v3.4.17-license-compliance.

 @chensun to see if this PR is ready to merge.

**Comment by hbelmiro** [comment]
/hold until all GitHub Actions tests pass to avoid accidentally merging with failing tests.
tide is not blocking the merge when GitHub checks are failing.

**Review by hbelmiro — COMMENTED** [review]
/lgtm

**Review by chensun — APPROVED** [review]
/lgtm
/approve

**Comment by gmfrasca** [comment]
I'm not sure why `kubeflow-pipelines-manifests` check seems to not initialize (eventually times out before test even begins), but if it gives any confidence, this same PR has passed the same check multiple times including last week according to the PROW history

**Pull request merged**